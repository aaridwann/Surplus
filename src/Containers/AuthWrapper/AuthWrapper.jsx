import React, { useEffect } from 'react';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { useSelector } from 'react-redux';
import LoginContainer from '../Login/Login.Container';
import { HomeStack } from '../../Navigation/Screens/Screen';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Constants from '../../Utils/Constants/Constants';
import ProfileContainer from '../ProfileContainer/ProfileContainer';
import RegistrationContainer from '../Registration/RegistrationContainer';

const AuthWrapper = () => {
	const { auth } = useSelector((state) => state);
	const { isLogin } = auth;
	const Stack = createNativeStackNavigator();

	useEffect(() => {}, [isLogin]);

	const Tab = createBottomTabNavigator();

	const StackNotLogin = () => (
		<Stack.Navigator>
			<Stack.Screen
				options={{ headerShown: false }}
				name="Login"
				component={LoginContainer}
			/>
			<Stack.Screen
				options={{ headerShown: false }}
				name="Registration"
				component={RegistrationContainer}
			/>
		</Stack.Navigator>
	);

	const StackAuthorize = () => {
		return (
			<Tab.Navigator
				screenOptions={({ route }) => ({
					tabBarIcon: ({ focused, color, size }) => {
						let iconName;
						if (route.name === 'Home') {
							iconName = focused ? 'home' : 'home-outline';
						} else if (route.name === 'Profile') {
							iconName = focused ? 'person-circle' : 'person-circle-outline';
						}

						// You can return any component that you like here!
						return <Ionicons name={iconName} size={size} color={color} />;
					},
					tabBarActiveTintColor: Constants.Colors.LightGreenSurplus,
					tabBarInactiveTintColor: 'gray',
				})}
			>
				<React.Fragment>
					<Tab.Screen
						key={'Home'}
						options={{ headerShown: false }}
						name={'Home'}
						component={HomeStack}
					/>
					<Tab.Screen
						key={'Profile'}
						options={{ headerShown: false }}
						name={'Profile'}
						component={ProfileContainer}
					/>
				</React.Fragment>
			</Tab.Navigator>
		);
	};

	return isLogin ? StackAuthorize() : StackNotLogin();
};
export default AuthWrapper;
