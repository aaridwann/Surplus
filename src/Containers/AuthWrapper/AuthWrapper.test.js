import React from 'react';
import { render } from '@testing-library/react-native';
import AuthWrapper from './AuthWrapper';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { useSelector } from 'react-redux';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

jest.mock('@react-navigation/native-stack', () => ({
	createNativeStackNavigator: jest.fn(),
}));

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
}));

jest.mock('../Login/Login.Container', () => 'LoginContainer');
jest.mock('../ProfileContainer/ProfileContainer', () => 'ProfileContainer');
jest.mock(
	'../Registration/RegistrationContainer',
	() => 'RegistrationContainer'
);
jest.mock('../../Navigation/Screens/Screen', () => ({
	HomeStack: 'HomeStack',
}));

jest.mock('@react-navigation/bottom-tabs', () => ({
	createBottomTabNavigator: jest.fn(),
}));

jest.mock('react-native-vector-icons/Ionicons', () => 'Ionicons');

describe('AuthWrapper', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('renders StackAuthorize when user is logged in', () => {
		useSelector.mockReturnValue({ auth: { isLogin: true } });

		const Tab = {
			Navigator: jest.fn(),
			Screen: jest.fn(),
		};

		createNativeStackNavigator.mockReturnValue(jest.fn());

		createBottomTabNavigator.mockReturnValue(Tab);

		const { toJSON } = render(<AuthWrapper />);
		expect(toJSON()).toMatchSnapshot();
	});

	test('renders StackNotLogin when user is not logged in', () => {
		useSelector.mockReturnValue({ auth: { isLogin: false } });

		createNativeStackNavigator.mockReturnValue(jest.fn());

		const { toJSON } = render(<AuthWrapper />);
		expect(toJSON()).toMatchSnapshot();
	});
});
