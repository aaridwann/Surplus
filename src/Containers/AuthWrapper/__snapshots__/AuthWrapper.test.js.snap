// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthWrapper renders StackAuthorize when user is logged in 1`] = `null`;
